system_tests_sasl_plain RouterTestVerifyHostNameNo::test_zzz_delete_create_ssl_profile `self.assertFalse(is_qdr_x)  # Should not be present now`

https://download.copr.fedorainfracloud.org/results/jdanek/skupper-router/fedora-36-ppc64le/03821395-skupper-router/builder-live.log.gz

```
27/64 Test #28: system_tests_sasl_plain .................***Failed   42.03 sec
test_inter_router_plain_exists (system_tests_sasl_plain.RouterTestPlainSasl)
Check authentication of inter-router link is PLAIN. ... ok
test_skstat_connect_sasl (system_tests_sasl_plain.RouterTestPlainSasl)
Make skstat use sasl plain authentication. ... ok
test_skstat_connect_sasl_password_file (system_tests_sasl_plain.RouterTestPlainSasl)
Make skstat use sasl plain authentication with client password specified in a file. ... ok
test_inter_router_sasl_fail (system_tests_sasl_plain.RouterTestPlainSaslFailure) ... ok
test_inter_router_sasl_fail (system_tests_sasl_plain.RouterTestPlainSaslFailureUsingLiteral) ... ok
test_aaa_skstat_connect_sasl_over_ssl (system_tests_sasl_plain.RouterTestPlainSaslOverSsl)
Make skstat use sasl plain authentication over ssl. ... ok
test_inter_router_plain_over_ssl_exists (system_tests_sasl_plain.RouterTestPlainSaslOverSsl)
The setUpClass sets up two routers with SASL PLAIN enabled over TLS. ... ok
test_inter_router_plain_over_ssl_exists (system_tests_sasl_plain.RouterTestVerifyHostNameNo)
Tests to make sure that an inter-router connection exists between the routers since verifyHostname is 'no'. ... ok
test_zzz_delete_create_ssl_profile (system_tests_sasl_plain.RouterTestVerifyHostNameNo)
Deletes a connector and its corresponding ssl profile and recreates both ... FAIL
test_no_inter_router_connection (system_tests_sasl_plain.RouterTestVerifyHostNameYes)
Tests to make sure that there are no 'inter-router' connections. ... ok

======================================================================
FAIL: test_zzz_delete_create_ssl_profile (system_tests_sasl_plain.RouterTestVerifyHostNameNo)
Deletes a connector and its corresponding ssl profile and recreates both
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/skupper-router/tests/system_tests_sasl_plain.py", line 721, in test_zzz_delete_create_ssl_profile
    self.assertFalse(is_qdr_x)  # Should not be present now
AssertionError: True is not false

----------------------------------------------------------------------
Ran 10 tests in 41.667s

FAILED (failures=1)
```

--- a/tests/system_tests_sasl_plain.py
+++ b/tests/system_tests_sasl_plain.py
@@ -20,7 +20,7 @@
 from time import sleep
 import os
 from subprocess import PIPE, Popen
-from system_test import TestCase, Qdrouterd, main_module, DIR, TIMEOUT
+from system_test import TestCase, Qdrouterd, main_module, DIR, TIMEOUT, retry_exception
 from system_test import unittest, QdManager, Process
 from skupper_router.management.client import Node
 from proton import SASL
@@ -716,9 +716,16 @@ class RouterTestVerifyHostNameNo(RouterTestPlainSaslCommon):
         self.assertIn("QDR.X", [c.container for c in connections])  # We can find the connection before
         local_node.delete(type='connector', name='connectorToX')
         local_node.delete(type='sslProfile', name='client-ssl-profile')
-        connections = local_node.query(type='io.skupper.router.connection').get_entities()
-        is_qdr_x = "QDR.X" in [c.container for c in connections]
-        self.assertFalse(is_qdr_x)  # Should not be present now
+
+        def check_connections():
+            conns = local_node.query(type='io.skupper.router.connection').get_entities()
+            is_qdr_x = "QDR.X" in [c.container for c in conns]
+            self.assertFalse(is_qdr_x)  # Should not be present now
+
+        # Even though we deleted the connector, it might take a bit of time for the actual connection
+        # object to be deleted since there might be some latency with the core thread. Retry the
+        # connection query till TIMEOUT seconds.
+        retry_exception(check_connections)
 
         # re-create the ssl profile
         local_node.create({'type': 'sslProfile',

